### PR TITLE
[#11099] Audit logs: restrict viewing and searching permissions to course owners

### LINKED ISSUE #11099: Audit logs: restrict viewing and searching permissions to course owners
**Description of task**

It could be better for only course owners to be allowed to access the audit logs feature, and not all instructors in general.
 
Both the frontend and backend will likely need to be changed, by restricting access to the `/web/instructor/logs` page and the backend logs searching endpoint.

### PR DESCRIPTION
**Pull request by AdithyaNarayan**
Fixes #11099 

**Outline of Solution**
- Added checks in backend for `Const.InstructorPermissions.CAN_MODIFY_STUDENT` (Enroll/Edit/Delete Students) as per discussion in issue.
- Added error message in frontend for insufficient permissions

Reminder: When the feature is launched, the link should be restricted in the instructor tab too.

### CODE REVIEW

**Review by halfwhole — COMMENTED** [review]
Could it be possible to limit the options in the course ID dropdown to only the ones that the instructor is allowed to access the logs for (using the `CAN_MODIFY_STUDENT` option)? That helps to prevent the user from making 'illegal' searches. Perhaps the `loadInstructorPrivilege` method could be useful here?

File: src/main/java/teammates/ui/webapi/GetFeedbackSessionLogsAction.java [hunk 1]
```diff
@@ -45,6 +45,12 @@ void checkSpecificAccessControl() {
 
         InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.getId());
         gateKeeper.verifyAccessible(instructor, courseAttributes);
```
Comment (t-cheepeng) [inline src/main/java/teammates/ui/webapi/GetFeedbackSessionLogsAction.java:47]: There is an overloaded method for the gatekeeper that accepts permission so you don't need to write the code below

File: src/web/app/pages-instructor/instructor-audit-logs-page/instructor-audit-logs-page.component.html [hunk 2]
```diff
@@ -72,6 +72,12 @@ <h1>Student Activity Logs</h1>
 </div>
 
 <div *tmIsLoading="isSearching">
+  <div *ngIf="searchErrorMessage" class="card card-default">
```
Comment (t-cheepeng) [inline src/web/app/pages-instructor/instructor-audit-logs-page/instructor-audit-logs-page.component.html:75]: Why not just use the status message service?

**Review by t-cheepeng — CHANGES_REQUESTED** [review]

**Comment by AdithyaNarayan** [comment]
@t-cheepeng @halfwhole Updated with recommended changes! Sorry for the delay

File: src/web/app/pages-instructor/instructor-audit-logs-page/instructor-audit-logs-page.component.spec.ts [hunk 3]
```diff
@@ -45,13 +45,33 @@ describe('InstructorAuditLogsPageComponent', () => {
...
+      canModifyCourse: true,
+      canModifySession: true,
+      canModifyStudent: true,
+      canModifyInstructor: true,
+      canViewStudentInSections: true,
+      canModifySessionCommentsInSections: true,
+      canViewSessionInSections: true,
+      canSubmitSessionInSections: true,
+    },
   };
```
Comment (halfwhole) [inline src/web/app/pages-instructor/instructor-audit-logs-page/instructor-audit-logs-page.component.spec.ts:75]: Should we also add a test course without `CAN_MODIFY_STUDENT` privileges, to ensure that it doesn't show up in the dropdown?

File: src/main/java/teammates/ui/webapi/GetFeedbackSessionLogsAction.java [hunk 4]
```diff
@@ -46,11 +46,7 @@ void checkSpecificAccessControl() {
...
```
[the remark this answers is not in the document]
Comment (t-cheepeng) [reply src/main/java/teammates/ui/webapi/GetFeedbackSessionLogsAction.java:47]: Yes, its the same

[the remark this answers is not in the document]
Comment (t-cheepeng) [reply src/main/java/teammates/ui/webapi/GetFeedbackSessionLogsAction.java:47]: Yes, it should be the same

**Review by halfwhole — APPROVED** [review]
LGTM, works well when tested. Could also be helpful to add comments specifying the new permission restrictions on the frontend, but the functionality looks good

**Review by t-cheepeng — APPROVED** [review]

**Review by madanalogy — CHANGES_REQUESTED** [review]
See https://github.com/TEAMMATES/teammates/issues/11099#issuecomment-848701508

**Comment by AdithyaNarayan** [comment]
Updated with the changes requested.

File: src/test/java/teammates/ui/webapi/GetFeedbackSessionLogsActionTest.java [hunk 5]
```diff
@@ -170,14 +170,28 @@ protected void testExecute() throws Exception {
...
+        InstructorAttributes helper = typicalBundle.instructors.get("helperOfCourse1");
         String courseId = instructor.getCourseId();
 
         ______TS("Only instructors of the same course can access");
         String[] submissionParams = new String[] {
                 Const.ParamsNames.COURSE_ID, courseId,
         };
         verifyOnlyInstructorsOfTheSameCourseCanAccess(submissionParams);
+
+        ______TS("Only instructors with modify student privilege can access");
```
Comment (madanalogy) [inline src/test/java/teammates/ui/webapi/GetFeedbackSessionLogsActionTest.java:183]:
```suggestion
        ______TS("Only instructors with co-owner or manager privilege can access");
```

**Comment by madanalogy** [comment]
@AdithyaNarayan remember to request a review when you're done, otherwise I might miss your changes haha

File: src/test/java/teammates/ui/webapi/GetFeedbackSessionLogsActionTest.java (hunk 5 above)
Comment (wkurniawan07) [reply src/test/java/teammates/ui/webapi/GetFeedbackSessionLogsActionTest.java:183]: @madanalogy sorry to disagree, but privilege and role cannot and should not be mixed. Current one is more accurate.

**Review by madanalogy — APPROVED** [review]
LGTM

**Review by wkurniawan07 — APPROVED** [review]

**Pull request merged**